MalmoEnv/malmoenv/commands.py

```diff
@@ -0,0 +1,229 @@
+# ------------------------------------------------------------------------------------------------
+# Copyright (c) 2018 Microsoft Corporation
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+# associated documentation files (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge, publish, distribute,
+# sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or
+# substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+# NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# ------------------------------------------------------------------------------------------------
+
+from lxml import etree
+import argparse
+
+
+class CommandHandlerException(Exception):
+    """For reporting errors in parsing Agent Command Handlers"""
+    def __init__(self, message):
+        self.message = message
+
+
+class CommandParser:
+    """Parse Agent Command Handlers to a list"""
+    continuousMovementCommands = "ContinuousMovement"
+    absoluteMovementCommands = "AbsoluteMovement"
+    discreteMovementCommands = "DiscreteMovement"
+    inventoryCommands = "Inventory"
+    chatCommands = "Chat"
+    simpleCraftCommands = "SimpleCraft"
+    missionQuitCommands = "MissionQuit"
+    humanLevelCommands = "HumanLevel"
+
+    ns = "{http://ProjectMalmo.microsoft.com}"
+
+    all_continuous = ["jump", "move", "pitch", "strafe", "turn", "crouch", "attack", "use"]
```

Comment:
Hoping to not have the XSDs bundled with the client (no MALMO_XSD_PATH) and overhead of parsing means this would have to be done at "build time". There are constants in the C code too. Could add item/issue to backlog.

Reply:
Yeah, it's non-urgent.